### PR TITLE
Rework non-code change detection, recombine master and branch unit test workflow

### PR DESCRIPTION
**Pull request by EnricoMi**
* removed code that gets changed files from pull requests, always get changed files for commit against default branch
* moved regexp patterns into Python script to have a single place with regexps
* allows to reuse the Python script in github workflow
* combines Master and Branch unit test workflow into a single one
* fixes different behaviour between buildkite and workflow: bildkite only runs tests when diff against master contains code changes, workflow only downloads tests when current commit has code changes (which is a different set of commits than buildkite tests)
* reduces complexity

### CODE REVIEW

**Comment by EnricoMi** [comment]
@tgaddair this is ready to go in, #2287 links here already

File: test/test_buildkite.py [hunk 1]
```diff
@@ -13,15 +13,19 @@
...
+import sys
 import unittest
 import warnings
 from shutil import copy
 
+from common import tempdir
 from horovod.runner.common.util import safe_shell_exec
 
-from common import tempdir
+sys.path.append('../.buildkite')
```
Comment (tgaddair) [inline test/test_buildkite.py:26]: This is assuming we're running out of the `test` folder.  To make it more generic, we could change it to:

```
sys.path.append(os.path.join(os.path.dirname(__file__), os.pardir, '.buildkite'))
```

**Review by tgaddair — COMMENTED** [review]
Looks good! Just one comment.

File: test/test_buildkite.py (hunk 1 above)
Comment (EnricoMi) [reply test/test_buildkite.py]: excellent, fixed and decorated with a comment

**Review by tgaddair — APPROVED** [review]
LGTM!  Thanks!

**Pull request merged**